Realm/ObjectServerTests/SwiftCollectionSyncTests.swift

```diff
@@ -338,3 +338,602 @@ class CollectionSyncTestCase: SwiftSyncTestCase {
+            object.stringCol = self.name
+            realm.add(object)
+        }
+
+        let realm = try await openRealm()
+        let results0 =  try await realm.objects(SwiftTypesSyncObject.self).where { $0.stringCol == name }.subscribe()
+        XCTAssertEqual(results0.count, 1)
+
+        let o = results0.first!
+
+        for (key, value) in dictionary {
+            assertDictionaryEqual(o, key, value)
+        }
+    }
+
+    @MainActor func testSyncMixedNestedArray() async throws {
+        let so = SwiftPerson()
+        so.firstName = "Doe"
+        let subArray3: AnyRealmValue = AnyRealmValue.fromArray([ .object(so), .double(123.456) ])
+        let subArray2: AnyRealmValue = AnyRealmValue.fromArray([ subArray3 ])
+        let subArray1: AnyRealmValue = AnyRealmValue.fromArray([ subArray2 ])
+        let array: Array<AnyRealmValue> = [
+            subArray1, .bool(false)
+        ]
+        try await write { realm in
+            let object = SwiftTypesSyncObject()
+            object.anyCol = AnyRealmValue.fromArray(array)
+            object.stringCol = self.name
+            realm.add(object)
+        }
+
+        let realm = try await openRealm()
+        let results0 =  try await realm.objects(SwiftTypesSyncObject.self).where { $0.stringCol == name }.subscribe()
+        XCTAssertEqual(results0.count, 1)
+
+        let o = results0.first!
+        XCTAssertEqual(o.anyCol.listValue?[1], .bool(false))
+        XCTAssertEqual(o.anyCol.listValue?[0].listValue?[0].listValue?[0].listValue?[1], .double(123.456))
+
+        XCTAssertEqual(o.anyCol.listValue?[0].listValue?[0].listValue?[0].listValue?[0].object(SwiftPerson.self)?.firstName, so.firstName)
+        XCTAssertEqual(o.anyCol.listValue?[1].boolValue, false)
+        XCTAssertEqual(o.anyCol.listValue?[0].listValue?[0].listValue?[0].listValue?[1].doubleValue, 123.456)
+    }
+
+    @MainActor func testSyncMixedNestedDictionary() async throws {
+        let so = SwiftPerson()
+        so.firstName = "Doe"
+        let subDict3: AnyRealmValue = AnyRealmValue.fromDictionary([ "key4": .object(so), "key5": .int(1202) ])
+        let subDict2: AnyRealmValue = AnyRealmValue.fromDictionary([ "key3": subDict3 ])
+        let subDict1: AnyRealmValue = AnyRealmValue.fromDictionary([ "key2": subDict2 ])
+        let dictionary: Dictionary<String, AnyRealmValue> = [
+            "key0": subDict1,
+            "key1": .bool(false)
+        ]
+        try await write { realm in
+            let object = SwiftTypesSyncObject()
+            object.anyCol = AnyRealmValue.fromDictionary(dictionary)
+            object.stringCol = self.name
+            realm.add(object)
+        }
+
+        let realm = try await openRealm()
+        let results0 =  try await realm.objects(SwiftTypesSyncObject.self).where { $0.stringCol == name }.subscribe()
+        XCTAssertEqual(results0.count, 1)
+
+        let o = results0.first!
+        XCTAssertEqual(o.anyCol.dictionaryValue?["key1"], .bool(false))
```

Comment:
Same. Using variables for common parts of the expressions would improve readability.